iac-extensions/docker/src/main/java/org/sonar/iac/docker/checks/RetrieveRemoteResourcesCheck.java

```diff
@@ -90,6 +172,32 @@ private static void checkArgument(CheckContext ctx, CommandDetector detector, Li
   }
 
   private static boolean containsWgetAuthenticationFlags(List<ArgumentResolution> args) {
-    return args.stream().anyMatch(arg -> WGET_AUTH_FLAGS.stream().anyMatch(flag -> arg.value().startsWith(flag)));
+    var containsSimpleAuthFlag = args.stream().anyMatch(arg -> WGET_AUTH_FLAGS.stream().anyMatch(flag -> arg.value().startsWith(flag)));
+    return containsSimpleAuthFlag || containsWgetAuthByHeader(args);
+  }
+
+  private static boolean containsWgetAuthByHeader(List<ArgumentResolution> args) {
+    return !WGET_AUTH_HEADERS_EQUALS.search(args).isEmpty() ||
+      !WGET_AUTH_HEADERS_SPACE.search(args).isEmpty();
+  }
+
+  private static void checkArgumentForCurl(CheckContext ctx, CommandDetector detector, List<ArgumentResolution> args) {
+    detector.search(args).forEach((CommandDetector.Command command) -> {
+      if (!containsCurlAuthenticationFlags(args)) {
+        var textRangeList = args.stream().map(a -> a.argument().textRange())
+          .toList();
+        var textRange = TextRanges.merge(textRangeList);
+        ctx.reportIssue(textRange, MESSAGE.formatted("curl"));
+      }
+    });
+  }
+
+  private static boolean containsCurlAuthenticationFlags(List<ArgumentResolution> args) {
+    var containsSimpleAuthFlag = args.stream().anyMatch(arg -> CURL_AUTH_FLAGS.stream().anyMatch(flag -> arg.value().startsWith(flag)));
+    return containsSimpleAuthFlag || containsCurlAuthByHeader(args);
+  }
+
+  private static boolean containsCurlAuthByHeader(List<ArgumentResolution> args) {
+    return !CURL_AUTH_HEADERS.search(args).isEmpty();
```

Comment:
We can see duplicated logic with what we have for `get`.
I believe it is fine, as there are not many identical duplicated lines.
Merging the common logic together would probably obfuscate the code with little benefits.

Reply:
Yes, it will complicate more the code.